### PR TITLE
fix(eslint-config): Ignore external modules with import/no-cycle

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Configures the `import/no-cycle` rule to ignore external modules to decrease the duration of the lint check.

In `@guardian/cdk` we're seeing `import/no-cycle` take a significant amount of time:

```console
➜ TIMING=1 npm run lint

> @guardian/cdk@47.1.0 lint
> eslint src --ext .ts

Rule                                    | Time (ms) | Relative
:---------------------------------------|----------:|--------:
import/no-cycle                         | 24660.951 |    81.0%
import/namespace                        |  3185.143 |    10.5%
prettier/prettier                       |  1226.525 |     4.0%
@typescript-eslint/no-misused-promises  |   283.835 |     0.9%
@typescript-eslint/no-floating-promises |   228.278 |     0.7%
@typescript-eslint/no-unsafe-assignment |   134.758 |     0.4%
@typescript-eslint/no-unsafe-argument   |   114.937 |     0.4%
@typescript-eslint/naming-convention    |   102.000 |     0.3%
import/no-unresolved                    |    76.968 |     0.3%
import/order                            |    69.877 |     0.2%
```

With `import/no-cycle` configured to ignore external modules, this is dramatically changed:

```console
➜ TIMING=1 npm run lint

> @guardian/cdk@47.1.0 lint
> eslint src --ext .ts

Rule                                    | Time (ms) | Relative
:---------------------------------------|----------:|--------:
import/namespace                        |  8576.203 |    54.0%
import/no-cycle                         |  4863.588 |    30.6%
prettier/prettier                       |  1130.410 |     7.1%
@typescript-eslint/no-misused-promises  |   278.362 |     1.8%
@typescript-eslint/no-floating-promises |   230.481 |     1.5%
@typescript-eslint/no-unsafe-assignment |   127.062 |     0.8%
@typescript-eslint/no-unsafe-argument   |   115.256 |     0.7%
@typescript-eslint/naming-convention    |    99.910 |     0.6%
import/order                            |    60.633 |     0.4%
import/no-unresolved                    |    45.641 |     0.3%
```

See: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-cycle.md#ignoreexternal

## How can we measure success?

Improved eslint performance.

### CODE REVIEW

File: libs/@guardian/eslint-config/.eslintrc.js [hunk 1]
```diff
@@ -75,6 +75,7 @@ module.exports = {
 		],
 
 		// prevent circular dependencies
-		'import/no-cycle': 2,
+		// See: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-cycle.md#ignoreexternal
```
Comment (mxdvl) [inline libs/@guardian/eslint-config/.eslintrc.js:78]: JSDoc maybe?

```suggestion
		/** @see https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-cycle.md#ignoreexternal */
```

**Review by mxdvl — APPROVED** [review]
This looks like a valuable improvement. If there are cyclical dependencies in external code, there’s not much we can do about it!

File: libs/@guardian/eslint-config/.eslintrc.js (hunk 1 above)
Comment (sndrs) [reply libs/@guardian/eslint-config/.eslintrc.js:78]: what will this change?

**Comment by sndrs** [comment]
> This looks like a valuable improvement. If there are cyclical dependencies in external code, there’s not much we can do about it!

agreed, i can't think of a good reason to not do this?

**Review by sndrs — CHANGES_REQUESTED** [review]
would you be able to [add a changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md)? you can run `make changeset`

File: libs/@guardian/eslint-config/.eslintrc.js (hunk 1 above)
Comment (akash1810) [reply libs/@guardian/eslint-config/.eslintrc.js]: I've dropped this change to be consistent with the rest of the file.

**Review by sndrs — APPROVED** [review]

**Pull request merged**